docs/fhir-guide/FHIR 101 - Practical Guide.ipynb

```diff
@@ -0,0 +1,2966 @@
+    "                \"id\": \"Patient.gender\",\n",
+    "                \"path\": \"Patient.gender\",\n",
+    "                \"min\": 1\n",
+    "            }\n",
+    "        ]\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "#### Anatomy of StructureDefinition"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "We won't cover this in great detail but here are the important parts of the `StructureDefinition`:\n",
+    "\n",
+    "- `resourceType` - every FHIR resource has to declare what type it is\n",
+    "- `baseDefinition` - this points to the `StructureDefinition` that the current one extends. It doesn't have to be one of the base FHIR definitions. It could be a `StructureDefinition`   from any FHIR data model. This is one of the features of FHIR that helps achieve interoperability.\n",
+    "- `url` - this is a required unique identifier for a `Conformance Resource`. It is a stable [canonical identifier](http://hl7.org/fhir/resource.html#canonical) for the resource and is never intended to change regardless of which FHIR server uses it.\n",
+    "- `fhirVersion` - the FHIR semantic version number is required. It will instruct  anything validating this resource to use R4 (4.0.0) validation logic.\n",
+    "- `differential` or `snapshot` - Every `StructureDefinition` must have either a `differential` - a section of content that specifies differences from the base definition, or a `snapshot` section which specifies the full definition. For model development, you almost always use the `differential` section and not the `snapshot`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "📘Learn more about [StructureDefinition](http://hl7.org/fhir/structuredefinition.html) anatomy\n",
+    "\n",
+    "📘Learn more about [Resource identifiers](http://hl7.org/fhir/resource.html#identification)\n",
+    "\n",
+    "📘Browse [FHIR data models](https://simplifier.net/search?searchtype=Projects) from other projects"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "### Remove Attributes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "We're going to start with the easiest modification first: removing an attribute from the base model. In fact, in our initial `Patient` `StructureDefinition`, we've already removed a handful of attributes. \n",
+    "\n",
+    "To remove an attribute you simply set its maximum cardinality to 0 in the `StructureDefinition` `differential` section: \n",
+    "\n",
+    "```json\n",
+    "{\n",
+    "    \"id\": \"Patient.name\",\n",
+    "    \"path\": \"Patient.name\",\n",
+    "    \"max\": \"0\"\n",
```

Comment:
Ah good question. I'll put something in the guide about that. 

I believe it's because `max` can take on values like `*` which means many. `min` can only take on a number. I tried looking and cannot find the documentation for this in the FHIR spec, but I've seen examples of it in various `StructureDefinition`s that have an attribute with a cardinality of `0..*`